digraph "unitGraph" {
    "r10 := @parameter0: java.lang.String[]"
    "$r0 = new java.lang.Throwable"
    "specialinvoke $r0.<java.lang.Throwable: void <init>(java.lang.String)>(\"because\")"
    "r1 = $r0"
    "$r2 = new java.lang.Throwable"
    "specialinvoke $r2.<java.lang.Throwable: void <init>(java.lang.Throwable)>(r1)"
    "r3 = $r2"
    "$r4 = new java.lang.TypeNotPresentException"
    "specialinvoke $r4.<java.lang.TypeNotPresentException: void <init>(java.lang.String,java.lang.Throwable)>(\"test\", r3)"
    "r5 = $r4"
    "throw r5"
    "$r6 := @caughtexception"
    "r7 = $r6"
    "$r8 = virtualinvoke r7.<java.lang.TypeNotPresentException: java.lang.Throwable getCause()>()"
    "if r3 == $r8 goto return"
    "$r9 = new java.lang.RuntimeException"
    "specialinvoke $r9.<java.lang.RuntimeException: void <init>(java.lang.String)>(\"Unexpected value of cause.\")"
    "throw $r9"
    "return"
    "r10 := @parameter0: java.lang.String[]"->"$r0 = new java.lang.Throwable";
    "$r0 = new java.lang.Throwable"->"specialinvoke $r0.<java.lang.Throwable: void <init>(java.lang.String)>(\"because\")";
    "specialinvoke $r0.<java.lang.Throwable: void <init>(java.lang.String)>(\"because\")"->"r1 = $r0";
    "r1 = $r0"->"$r2 = new java.lang.Throwable";
    "$r2 = new java.lang.Throwable"->"specialinvoke $r2.<java.lang.Throwable: void <init>(java.lang.Throwable)>(r1)";
    "specialinvoke $r2.<java.lang.Throwable: void <init>(java.lang.Throwable)>(r1)"->"r3 = $r2";
    "r3 = $r2"->"$r4 = new java.lang.TypeNotPresentException";
    "$r4 = new java.lang.TypeNotPresentException"->"specialinvoke $r4.<java.lang.TypeNotPresentException: void <init>(java.lang.String,java.lang.Throwable)>(\"test\", r3)";
    "specialinvoke $r4.<java.lang.TypeNotPresentException: void <init>(java.lang.String,java.lang.Throwable)>(\"test\", r3)"->"r5 = $r4";
    "r5 = $r4"->"throw r5";
    "r5 = $r4"->"$r6 := @caughtexception";
    "throw r5"->"$r6 := @caughtexception";
    "$r6 := @caughtexception"->"r7 = $r6";
    "r7 = $r6"->"$r8 = virtualinvoke r7.<java.lang.TypeNotPresentException: java.lang.Throwable getCause()>()";
    "$r8 = virtualinvoke r7.<java.lang.TypeNotPresentException: java.lang.Throwable getCause()>()"->"if r3 == $r8 goto return";
    "if r3 == $r8 goto return"->"$r9 = new java.lang.RuntimeException";
    "if r3 == $r8 goto return"->"return";
    "$r9 = new java.lang.RuntimeException"->"specialinvoke $r9.<java.lang.RuntimeException: void <init>(java.lang.String)>(\"Unexpected value of cause.\")";
    "specialinvoke $r9.<java.lang.RuntimeException: void <init>(java.lang.String)>(\"Unexpected value of cause.\")"->"throw $r9";
}
